projects/com.oracle.truffle.llvm/src/com/oracle/truffle/llvm/LLVM.java

```diff
@@ -104,6 +144,20 @@ public static void main(String[] args) throws Exception {
         System.exit(status);
     }
 
+    public static ParserResult parseString(String source, LLVMContext context) throws IOException {
+        final File file = File.createTempFile("sulong", ".ll");
+
+        try {
+            try (FileOutputStream stream = new FileOutputStream(file.getPath())) {
+                stream.write(source.getBytes(StandardCharsets.UTF_8));
+            }
+
+            return parseFile(file.getPath(), context);
+        } finally {
+            file.delete();
+        }
+    }
+
```

Comment:
Couldn't figure out how to run the parser on a string without writing it out to a file.